charts/kargo/templates/controller/deployment.yaml

```diff
@@ -27,9 +27,11 @@ spec:
         imagePullPolicy: {{ .Values.image.pullPolicy }}
         command: ["/usr/local/bin/kargo", "controller"]
         env:
-        {{- if .Values.controller.shardName }}
+        - name: IS_DEFAULT_CONTROLLER
+          value: {{ quote .Values.controller.sharding.isDefault }}
+        {{- if .Values.controller.sharding.shardName }}
         - name: SHARD_NAME
-          value: {{ .Values.controller.shardName }}
+          value: {{ .Values.controller.sharding.shardName }}
```

Comment:
don't think we need another env, we can probably just assume that if shard name is not set it will act on unlabeled stage. I doubt there will be a situation where a user will label the stage without having a labeled shard controller and even if they do that's a mistake that can cause problems down the line so better to just ignore those completely keeping it consistent. wdyt?

Reply:
Prior to this PR, we could set a controller to reconcile _all_ Stages and Promotions (i.e. no sharding at all) or all _for a specific shard_.

I thought I was hearing at the meeting this morning that this was insufficient because in cases where there _is_ sharding, we may wish to designate a controller as the default, thereby reconciling both resources for its own shard _and_ unlabeled resource.

Did I misunderstand?

Reply:
Ok. Got it. I will amend this accordingly.

Reply:
As Jesse mentioned I too think, both should be mutually exclusive.